File: Python/programmers/etc/level2/조이스틱.py
def solution(name):
    answer = 0
    _min = len(name)-1
    for idx, s in enumerate(name):
        answer += min(ord(s)-ord("A"), ord("Z")-ord(s)+1)
        nxt = idx+1
        while nxt < len(name) and name[nxt] == "A":
            nxt += 1
        _min = min(_min, 2*idx+len(name)-nxt, idx+2*(len(name)-nxt))
    answer += _min
    return answer


name = "JEROEN"
print(solution(name)) # 56

name = "JAN"
print(solution(name)) # 23

File: Java/programmers/etc/level2/조이스틱.java
package CodeTest.Java.programmers.etc.level2;

public class 조이스틱 {
    public int solution(String name) {
        int answer = 0;
        int _min = name.length()-1;
        
        int nxt = 0;
        
        for (int idx=0; idx<name.length(); idx++) {
            char c = name.charAt(idx);
            answer += Math.min(c-'A', 'Z'-c+1);
            nxt = idx+1;
            while (nxt < name.length() && name.charAt(nxt) == 'A') {
                nxt++;
            }
            _min = Math.min(_min, Math.min(2*idx+name.length()-nxt, idx+2*(name.length()-nxt)));
        }
        answer += _min;

        return answer;
    }
}
